### PR TITLE
Update to filecoin 0.5.6

### PR DESCRIPTION
**Pull request by raynaudoe**
- Fixed compatibility issues
- Added filecoin-ffi as git module (workaround until it gets detached from lotus api)
- Added new gas parameters: `gasLimit`, `gasPremium`, `gasFeeCap`
- Fixed integrations tests

### CODE REVIEW

File: rosetta/services/construction.go [hunk 1]
```diff
@@ -29,14 +34,18 @@ const OptionsBlockInclKey = "blockIncl"
 // ConstructionMetadataResponse that specifies the next valid nonce.
 const NonceKey = "nonce"
 
-// GasPriceKey is the name of the key in the Metadata map inside a
+// GasPremiumKey is the name of the key in the Metadata map inside a
 // ConstructionMetadataResponse that specifies tx's gas price
```
Comment (jleni) [inline rosetta/services/construction.go:38]:
```suggestion
// ConstructionMetadataResponse that specifies tx's gas premium
```

File: rosetta/examples/create_transaction.go [hunk 2]
```diff
@@ -72,9 +72,10 @@ func main() {
 	}
 
 	mtx := rosettaFilecoinLib.TxMetadata{
-		Nonce:    uint64(respMetadata.Metadata[services.NonceKey].(float64)),
-		GasPrice: uint64(respMetadata.Metadata[services.GasPriceKey].(float64)),
-		GasLimit: gasLimit,
+		Nonce:      uint64(respMetadata.Metadata[services.NonceKey].(float64)),
+		GasPremium: uint64(respMetadata.Metadata[services.GasPremiumKey].(float64)),
```
Comment (jleni) [inline rosetta/examples/create_transaction.go:76]: We will have a problem here..
`GasPremium` and `GasFeeCap` are `big.Int` so float64 may not be enough to represent the full value.
I think we should work with strings here instead

File: rosetta/examples/create_transaction.go [hunk 3]
```diff
@@ -72,9 +72,10 @@ func main() {
...
+		GasFeeCap:  uint64(respMetadata.Metadata[services.GasFeeCapKey].(float64)),
```
Comment (jleni) [inline rosetta/examples/create_transaction.go:77]: Same as above

File: rosetta/examples/create_transaction.go [hunk 4]
```diff
@@ -72,9 +72,10 @@ func main() {
...
+		GasLimit:   gasLimit,
```
Comment (jleni) [inline rosetta/examples/create_transaction.go:78]: We can now remove this

File: rosetta/services/construction.go [hunk 5]
```diff
@@ -57,75 +66,100 @@ func (c *ConstructionAPIService) ConstructionMetadata(
...
 	)
 
 	errNet := ValidateNetworkId(ctx, &c.node, request.NetworkIdentifier)
 	if errNet != nil {
 		return nil, errNet
 	}
 
-	gasLimit := filTypes.NewInt(uint64(build.BlockGasLimit))
 	md := make(map[string]interface{})
+	gasLimit := int64(build.BlockGasLimit)
```
Comment (jleni) [inline rosetta/services/construction.go:82]: I think this is obsolete

File: rosetta/services/construction.go [hunk 6]
```diff
@@ -57,75 +66,100 @@ func (c *ConstructionAPIService) ConstructionMetadata(
...
+			var gasErr error
+			message := &filTypes.Message{From: addressSenderParsed}
+			// GasEstimateGasLimit requires receiver address
+			if okReceiver {
+				addressReceiverParsed, err := address.NewFromString(addressReceiverRaw.(string))
+				if err != nil {
+					return nil, ErrInvalidAccountAddress
+				}
+				message.To = addressReceiverParsed
+				gasLimit, gasErr = c.node.GasEstimateGasLimit(ctx, message, filTypes.TipSetKey{})
```
Comment (jleni) [inline rosetta/services/construction.go:131]: This sounds obsolete now

File: rosetta/services/mocks/fullnode_mock.go [hunk 7]
```diff
@@ -22,6 +25,170 @@ type FullNodeMock struct {
...
+
+func (f *FullNodeMock) BeaconGetEntry(ctx context.Context, epoch abi.ChainEpoch) (*filTypes.BeaconEntry, error) {
+	panic("implement me")
+}
+
+func (f *FullNodeMock) GasEstimateFeeCap(ctx context.Context, message *filTypes.Message, i int64, key filTypes.TipSetKey) (filTypes.BigInt, error) {
+	panic("implement me")
+}
+
+func (f *FullNodeMock) GasEstimateGasLimit(ctx context.Context, message *filTypes.Message, key filTypes.TipSetKey) (int64, error) {
```
Comment (jleni) [inline rosetta/services/mocks/fullnode_mock.go:88]: We can probably delete this

**Review by jleni — CHANGES_REQUESTED** [review]

File: rosetta/examples/create_transaction.go (hunk 2 above)
Comment (raynaudoe) [reply rosetta/examples/create_transaction.go]: Ok, fixed it, this was outdated since gas data comes originally on string format.

**Pull request merged**